topology/nhlt/intel/dmic/dmic-process.c

```diff
@@ -0,0 +1,1341 @@
+	*sensitivity = dmic->dmic_mic_config.sensitivity;
+
+	return 0;
+}
+
+int dmic_get_mic_params(struct intel_nhlt_params *nhlt, int index,
+			uint8_t *type, uint8_t *panel, uint32_t *speaker_position_distance,
+			uint32_t *horizontal_offset, uint32_t *vertical_offset,
+			uint8_t *frequency_low_band, uint8_t *frequency_high_band,
+			uint16_t *direction_angle, uint16_t *elevation_angle,
+			uint16_t *vertical_angle_begin, uint16_t *vertical_angle_end,
+			uint16_t *horizontal_angle_begin, uint16_t *horizontal_angle_end)
+{
+	struct intel_dmic_params *dmic = (struct intel_dmic_params *)nhlt->dmic_params;
+
+	if (!dmic)
+		return -EINVAL;
+
+	*type = dmic->dmic_mic_config.vendor[index].type;
+	*panel = dmic->dmic_mic_config.vendor[index].panel;
+	*speaker_position_distance = dmic->dmic_mic_config.vendor[index].speaker_position_distance;
+	*horizontal_offset = dmic->dmic_mic_config.vendor[index].horizontal_offset;
+	*vertical_offset = dmic->dmic_mic_config.vendor[index].vertical_offset;
+	*frequency_low_band = dmic->dmic_mic_config.vendor[index].frequency_low_band;
+	*frequency_high_band = dmic->dmic_mic_config.vendor[index].frequency_high_band;
+	*direction_angle = dmic->dmic_mic_config.vendor[index].direction_angle;
+	*elevation_angle = dmic->dmic_mic_config.vendor[index].elevation_angle;
+	*vertical_angle_begin = dmic->dmic_mic_config.vendor[index].vertical_angle_begin;
+	*vertical_angle_end = dmic->dmic_mic_config.vendor[index].vertical_angle_end;
+	*horizontal_angle_begin = dmic->dmic_mic_config.vendor[index].horizontal_angle_begin;
+	*horizontal_angle_end = dmic->dmic_mic_config.vendor[index].horizontal_angle_end;
+
+	return 0;
+}
+
+int dmic_get_vendor_blob_size(struct intel_nhlt_params *nhlt, size_t *size)
+{
+	struct intel_dmic_params *dmic = (struct intel_dmic_params *)nhlt->dmic_params;
+	int i, fir_index_0, fir_index_1;
+
+	if (!dmic || !dmic->dmic_count)
+		return -EINVAL;
+
+	*size = sizeof(struct dmic_intel_config_data);
+
+	/* if either of the fir is 0 length, copy the existing fir twice */
+	fir_index_0 = 0;
+	fir_index_1 = 1;
+	if (dmic->dmic_fir_array.fir_len[0] == 0) {
+		fir_index_0 = 1;
+		fir_index_1 = 1;
+	}
+	if (dmic->dmic_fir_array.fir_len[1] == 0) {
+		fir_index_0 = 0;
+		fir_index_1 = 0;
+	}
+
+	/* variable amount of pdm's */
+	for (i = 0; i < DMIC_HW_CONTROLLERS; i++) {
+		/* only copy the pdm data if it is enabled */
+		if ((dmic->dmic_blob.channel_pdm_mask & BIT(i)) == 0)
+			continue;
+
+		*size += sizeof(struct dmic_intel_pdm_ctrl_cfg);
+		*size += sizeof(struct dmic_intel_fir_config) * DMIC_HW_FIFOS;
+
+		*size += dmic->dmic_fir_array.fir_len[fir_index_0] * sizeof(uint32_t);
+		*size += dmic->dmic_fir_array.fir_len[fir_index_1] * sizeof(uint32_t);
+	}
+
+	return 0;
+}
+
+int dmic_get_vendor_blob_count(struct intel_nhlt_params *nhlt)
+{
+	struct intel_dmic_params *dmic = (struct intel_dmic_params *)nhlt->dmic_params;
+
+	if (!dmic || !dmic->dmic_count)
+		return -EINVAL;
+
+	return dmic->dmic_count;
```

Comment:
same comment as above, not sure how 'vendor_blob_count' and 'dmic_count' are identical?